### PR TITLE
hwmv2: Port arm cypress socs/boards

### PR DESCRIPTION
**Pull request by nordicjm**

### CODE REVIEW

File: boards/cypress/cy8ckit_062_wifi_bt/cy8ckit_062_wifi_bt_cy8c6247_m4_defconfig [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
 #
 
-CONFIG_SOC_SERIES_PSOC62=y
-CONFIG_SOC_PSOC6_M4=y
-CONFIG_SOC_PART_NUMBER_CY8C6247BZI_D54=y
-CONFIG_BOARD_CY8CKIT_062_WIFI_BT_M4=y
+#CONFIG_SOC_SERIES_PSOC62=y
+#CONFIG_SOC_PSOC6_M4=y
+#CONFIG_SOC_PART_NUMBER_CY8C6247BZI_D54=y
+#CONFIG_BOARD_CY8CKIT_062_WIFI_BT_M4=y
```
Comment (gmarull) [inline boards/cypress/cy8ckit_062_wifi_bt/cy8ckit_062_wifi_bt_cy8c6247_m4_defconfig:10]: ?

File: soc/cypress/Kconfig.soc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+config SOC_FAMILY_INFINEON_CAT1
+	bool
+
+config SOC_FAMILY_INFINEON_CAT1A
+	bool
+
+config SOC_FAMILY
+	default "psoc6" if SOC_FAMILY_PSOC6
+	default "infineon_cat1" if SOC_FAMILY_INFINEON_CAT1
```
Comment (gmarull) [inline soc/cypress/Kconfig.soc:15]: what about CAT1A?

Comment (nordicjm) [reply soc/cypress/Kconfig.soc:15]: This soc was created in a mess, cat and cat1a are always selected, it does something in their HAL

**Review by gmarull — CHANGES_REQUESTED** [review]
update docs/maintainers as needed

File: boards/cypress/cy8ckit_062_ble/Kconfig.cy8ckit_062_ble [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	select SOC_CY8C6347_M0
+	help
+	  PSoC6 BLE Pioneer Kit [M0 CPU0]
+
+config BOARD_CY8CKIT_062_BLE_CY8C6347_M4
+	bool
+	select SOC_PART_NUMBER_CY8C6347BZI_BLD53
+	select SOC_CY8C6347_M4
+	help
+	  PSoC6 BLE Pioneer Kit [M4 CPU1]
```
Comment (tejlmand) [inline boards/cypress/cy8ckit_062_ble/Kconfig.cy8ckit_062_ble:19]:
```suggestion
config BOARD_CY8CKIT_062_BLE
	select SOC_CY8C6347_M0 if BOARD_CY8CKIT_062_BLE_CY8C6347_M0
	select SOC_CY8C6347_M4 if BOARD_CY8CKIT_062_BLE_CY8C6347_M4
	select SOC_PART_NUMBER_CY8C6347BZI_BLD53
```

File: boards/cypress/cy8ckit_062_wifi_bt/Kconfig.cy8ckit_062_wifi_bt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	select SOC_CY8C6247_M0
+	help
+	  PSoC6 WiFi-BT Pioneer Kit M0
+
+config BOARD_CY8CKIT_062_WIFI_BT_CY8C6247_M4
+	bool
+	select SOC_PART_NUMBER_CY8C6247BZI_D54
+	select SOC_CY8C6247_M4
+	help
+	  PSoC6 WiFi-BT Pioneer Kit M4
```
Comment (tejlmand) [inline boards/cypress/cy8ckit_062_wifi_bt/Kconfig.cy8ckit_062_wifi_bt:19]:
```suggestion
config BOARD_CY8CKIT_062_WIFI_BT_CY8C6247_M0
	select SOC_PART_NUMBER_CY8C6247BZI_D54
	select SOC_CY8C6247_M0 if BOARD_CY8CKIT_062_WIFI_BT_CY8C6247_M0
	select SOC_CY8C6247_M4 if BOARD_CY8CKIT_062_WIFI_BT_CY8C6247_M4
```

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: boards/cypress/cy8ckit_062_ble/Kconfig.cy8ckit_062_ble (hunk 3 above)
Comment (nordicjm) [reply boards/cypress/cy8ckit_062_ble/Kconfig.cy8ckit_062_ble]: Fixed

File: boards/cypress/cy8ckit_062_wifi_bt/Kconfig.cy8ckit_062_wifi_bt (hunk 4 above)
Comment (nordicjm) [reply boards/cypress/cy8ckit_062_wifi_bt/Kconfig.cy8ckit_062_wifi_bt]: Fixed

**Review by tejlmand — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**